### PR TITLE
[feature] piv: addition of service

### PR DESCRIPTION
**Pull request by patochectp**
ref: ND-1083

### CODE REVIEW

**Review by woshilapin — APPROVED** [review]

**Review by woshilapin — APPROVED** [review]

File: tests/integration/piv_test.py [hunk 1]
```diff
@@ -232,6 +232,217 @@ def _get_stomp_20201022_23187_delayed_5min_fixture():
...
+            "dateHeureReelle": "2020-10-22T22:22:00+02:00",
+            "sourceHoraire": "CO",
+            "typeAffichage": "HORAIRE",
+            "indicateurAdaptation": False,
+            "planTransportSource": "PTA",
+        },
+        "depart": {
+            "evenement": {
+                "texte": "",
+                "type": "CREATION",
```
Comment (pbougue) [inline tests/integration/piv_test.py:278]: As discussed : We should pass some into `statutModification` instead... But it requires some adds in ModelBuilder too.

File: tests/integration/piv_test.py [hunk 2]
```diff
@@ -1045,3 +1256,115 @@ def test_piv_full_back_ok_on_stops(mock_rabbitmq):
...
+        assert added_st.departure == datetime(2020, 10, 22, 20, 24)
+
+        added_st = db_trip.stop_time_updates[2]
+        assert added_st.stop_id == "stop_point:PIV:85010231:Train"
+        assert added_st.arrival_status == ModificationType.add.name
+        assert added_st.arrival == datetime(2020, 10, 22, 20, 32)
+        assert added_st.departure_status == ModificationType.none.name
+        assert added_st.departure == datetime(2020, 10, 22, 20, 34)
+
+        for s in db_trip.stop_time_updates[3:7]:
```
Comment (pbougue) [inline tests/integration/piv_test.py:1297]:
```suggestion
        for s in db_trip.stop_time_updates[3:]:
```

File: tests/integration/piv_test.py [hunk 3]
```diff
@@ -1045,3 +1256,115 @@ def test_piv_full_back_ok_on_stops(mock_rabbitmq):
...
+        db_trip = TripUpdate.query.first()
+
+        assert db_trip.status == "update"
+        assert db_trip.effect == "MODIFIED_SERVICE"
+        assert db_trip.company_id == "company:PIVPP:1187"
+        assert db_trip.physical_mode_id is None
+        assert db_trip.headsign == "23187"
+        assert len(db_trip.stop_time_updates) == 7
+
+        for s in db_trip.stop_time_updates[0:6]:
```
Comment (pbougue) [inline tests/integration/piv_test.py:1319]:
```suggestion
        for s in db_trip.stop_time_updates:
```

File: tests/integration/piv_test.py [hunk 4]
```diff
@@ -1045,3 +1256,115 @@ def test_piv_full_back_ok_on_stops(mock_rabbitmq):
...
+        db_trip = TripUpdate.query.first()
+
+        assert db_trip.status == "update"
+        assert db_trip.effect == "MODIFIED_SERVICE"
+        assert db_trip.company_id == "company:PIVPP:1187"
+        assert db_trip.physical_mode_id is None
+        assert db_trip.headsign == "23187"
+        assert len(db_trip.stop_time_updates) == 8
+
+        for s in db_trip.stop_time_updates[0:4]:
```
Comment (pbougue) [inline tests/integration/piv_test.py:1349]:
```suggestion
        for s in db_trip.stop_time_updates[:4]:
```

**Review by pbougue — COMMENTED** [review]
Nice tests, and well spotted for `statutModification`.
I'm OK to do that in another PR if it looks better.

File: tests/integration/piv_test.py [hunk 5]
```diff
@@ -51,6 +52,9 @@
 from tests import mock_navitia
 from tests.integration.conftest import PIV_CONTRIBUTOR_ID
 
+ModificationTuple = namedtuple("ModificationTuple", ["statut", "motif"])
+DisruptionTuple = namedtuple("DisruptionTuple", ["type", "texte"])
```
Comment (pbougue) [inline tests/integration/piv_test.py:56]: :candy: neat !

File: tests/integration/piv_test.py [hunk 6]
```diff
@@ -232,6 +253,180 @@ def _get_stomp_20201022_23187_delayed_5min_fixture():
...
+        },
+        "depart": {
+            "numeroCirculation": "23187",
+            "dateHeure": "2020-10-22T23:31:00+02:00",
+            "dateHeureReelle": "2020-10-22T23:31:00+02:00",
+            "sourceHoraire": "CO",
+            "typeAffichage": "HORAIRE",
+            "indicateurAdaptation": False,
+            "planTransportSource": "PTA",
+            "statutModification": "CREATION",
```
Comment (pbougue) [inline tests/integration/piv_test.py:399]: minor: It's not listed in examples we had, but we may test with some in "evenement.type" to check a bit more compliance with specs (I have not seen any, but it wan be my eyes :stuck_out_tongue:).

File: kirin/piv/model_maker.py [hunk 7]
```diff
@@ -388,29 +390,25 @@ def _make_trip_update(self, json_train, vj):
...
                 if not piv_event_status or piv_event_status in MANAGED_STOP_EVENTS:
                     piv_event_datetime = get_value(event, "dateHeureReelle", nullable=True)
                     event_datetime = as_utc_naive_dt(piv_event_datetime) if piv_event_datetime else None
                     if event_datetime:
                         rt_stop_time[event_toggle] = event_datetime
                         setattr(st_update, STOP_EVENT_DATETIME_MAP[event_toggle], event_datetime)
-                    if piv_disruption:
+                    if piv_disruption and piv_event_status in ["RETARD_OBSERVE", "RETARD_PROJETE"]:
                         piv_event_delay = piv_disruption.get("retard", {}).get("duree", 0)
                         setattr(st_update, DELAY_MAP[event_toggle], as_duration(piv_event_delay * 60))  # minutes
```
Comment (pbougue) [inline kirin/piv/model_maker.py:404]: minor: I would separate both (and too bad if it's not consistent in the feed)
```suggestion
                    if piv_disruption:
                        piv_event_delay = piv_disruption.get("retard", {}).get("duree", 0)

                    if piv_event_status in ["RETARD_OBSERVE", "RETARD_PROJETE"]:
                        setattr(st_update, DELAY_MAP[event_toggle], as_duration(piv_event_delay * 60))  # minutes
```

**Review by pbougue — APPROVED** [review]

File: kirin/piv/model_maker.py (hunk 7 above)
Comment (patochectp) [reply kirin/piv/model_maker.py:404]: if not piv_disruption and  piv_event_status in ["RETARD_OBSERVE", "RETARD_PROJETE"], `piv_event_delay` will be not referenced

File: tests/integration/piv_test.py (hunk 6 above)
Comment (patochectp) [reply tests/integration/piv_test.py:411]: done in dd91e4cdfeb16f95bb3ded97af079d49f44bb5f1

File: kirin/piv/model_maker.py (hunk 7 above)
Comment (patochectp) [reply kirin/piv/model_maker.py]: done in c0ebe4d0a59968277aaf7de7e7a45fe34911248b

**Review by pbougue — APPROVED** [review]
Those tests enhancements are really nice :heart_eyes:

**Pull request merged**